tools/src/bin/collect.rs

```diff
@@ -450,31 +447,78 @@ struct Options {
 }
 
 impl Options {
-    fn hpke_keypair(&self) -> Result<HpkeKeypair, anyhow::Error> {
+    fn collector_credential(&self) -> Result<Option<PrivateCollectorCredential>, Error> {
         match (
-            &self.hpke_config.hpke_config,
-            &self.hpke_config.hpke_private_key,
-            &self.hpke_config.collector_credential_file,
             &self.hpke_config.collector_credential,
+            &self.hpke_config.collector_credential_file,
         ) {
-            (Some(config), Some(private), _, _) => {
-                Ok(HpkeKeypair::new(config.clone(), private.clone()))
-            }
-            (None, None, Some(collector_credential_file), _) => {
+            (Some(collector_credential), _) => Ok(Some(collector_credential.clone())),
```

Comment:
nit: specifying a collector credential directly is mutually-exclusive with specifying a collector credential file (enforced by Clap config); this code might read a bit easier if the patterns were written such that they match only if exactly one of the given options is specified, with a default match arm panicking with some useful error message.

Same issue with the match block in the `authentication_token` function below. While touching things, consider updating the `unreachable!` arm in `hpke_keypair` to also panic with a useful message.